nvm.sh

```diff
@@ -2106,19 +2106,31 @@ nvm_die_on_prefix() {
   fi
 
   local NVM_NPM_PREFIX
-  NVM_NPM_PREFIX="$(npm config --loglevel=warn get prefix)"
-  if ! (nvm_tree_contains_path "$NVM_DIR" "$NVM_NPM_PREFIX" >/dev/null 2>&1); then
-    if [ "_$NVM_DELETE_PREFIX" = "_1" ]; then
-      npm config --loglevel=warn delete prefix
-    else
-      nvm deactivate >/dev/null 2>&1
-      nvm_err "nvm is not compatible with the npm config \"prefix\" option: currently set to \"$NVM_NPM_PREFIX\""
-      if nvm_has 'npm'; then
-        nvm_err "Run \`npm config delete prefix\` or \`$NVM_COMMAND\` to unset it."
+  local npmrcs=". $HOME"
+
+  for npmrc in $npmrcs; do
+    if [ -f "${npmrc}/.npmrc" ]; then
```

Comment:
what I do like about this PR in particular is adding this `npmrc` check.

What would you think about adding this check *prior to* the `npm config get prefix` check? That would enable users to get a faster warning when things were broken, pointing them at the exact file that needed to be modified - and it'd still go through the `npm config` check if this grep passed.

That would also serve as a more iterative approach towards removing or bypassing the config check altogether (which I haven't agreed to yet).

Reply:
Well yes, but in that case if I understand your intention right `npm config get prefix` would be called for every start again. Then we're were we started. But as additional check yes, that could tell the user which config is the nasty one.

Reply:
Yes; and it'd be a further thing that wouldn't have to be removed or bypassed even if the `npm config` call was.

Reply:
In that case this information is alrready there as we can just set a variable to the current npmrc value before the break statement and use that below in the error message.

Reply:
So added the suggested error here https://github.com/creationix/nvm/pull/1597/files#diff-e1e8e036966ad5aaf7709712cd5d8ee3R2115